l1-contracts/contracts/bridge/ntv/L2NativeTokenVault.sol

```diff
@@ -0,0 +1,215 @@
+    }
+
+    /// @notice Deploys the beacon proxy for the L2 token, while using ContractDeployer system contract.
+    /// @dev This function uses raw call to ContractDeployer to make sure that exactly `l2TokenProxyBytecodeHash` is used
+    /// for the code of the proxy.
+    /// @param _salt The salt used for beacon proxy deployment of L2 bridged token.
+    /// @return proxy The beacon proxy, i.e. L2 bridged token.
+    function _deployBeaconProxy(bytes32 _salt) internal override returns (BeaconProxy proxy) {
+        if (address(L2_LEGACY_SHARED_BRIDGE) == address(0)) {
+            // Deploy the beacon proxy for the L2 token
+
+            (bool success, bytes memory returndata) = SystemContractsCaller.systemCallWithReturndata(
+                uint32(gasleft()),
+                DEPLOYER_SYSTEM_CONTRACT,
+                0,
+                abi.encodeCall(
+                    IContractDeployer.create2,
+                    (_salt, l2TokenProxyBytecodeHash, abi.encode(address(bridgedTokenBeacon), ""))
+                )
+            );
+
+            // The deployment should be successful and return the address of the proxy
+            if (!success) {
+                revert DeployFailed();
+            }
+            proxy = BeaconProxy(abi.decode(returndata, (address)));
+        } else {
+            // Deploy the beacon proxy for the L2 token
+            address l2TokenAddr = L2_LEGACY_SHARED_BRIDGE.deployBeaconProxy(_salt);
+            proxy = BeaconProxy(payable(l2TokenAddr));
+        }
+    }
+
+    function _withdrawFunds(bytes32 _assetId, address _to, address _token, uint256 _amount) internal override {
+        if (_assetId == BASE_TOKEN_ASSET_ID) {
+            revert AssetIdNotSupported(BASE_TOKEN_ASSET_ID);
+        } else {
+            // Withdraw funds
+            IERC20(_token).safeTransfer(_to, _amount);
+        }
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            INTERNAL & HELPER FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Calculates L2 wrapped token address given the currently stored beacon proxy bytecode hash and beacon address.
+    /// @param _l1Token The address of token on L1.
+    /// @return Address of an L2 token counterpart.
+    function calculateCreate2TokenAddress(
+        uint256 _originChainId,
+        address _l1Token
+    ) public view override(INativeTokenVault, NativeTokenVault) returns (address) {
+        bytes32 constructorInputHash = keccak256(abi.encode(address(bridgedTokenBeacon), ""));
+        bytes32 salt = _getCreate2Salt(_originChainId, _l1Token);
+        address deployerAddress = address(L2_LEGACY_SHARED_BRIDGE) == address(0)
+            ? address(this)
+            : address(L2_LEGACY_SHARED_BRIDGE);
+        return
+            L2ContractHelper.computeCreate2Address(
+                deployerAddress,
+                salt,
+                l2TokenProxyBytecodeHash,
```

Comment:
here you use the same `l2TokenProxyBytecodeHash` as was used during the deployment of the chain. however, when legacy bridge is used for deployment, the token proxy bytecodehash that will be used is the same as in the legacy bridge.

You should probably call the legacy bridge to get an address